src/platform/intel/cavs/lps_pic_restore_vector.S

```diff
@@ -0,0 +1,108 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// Copyright(c) 2019 Intel Corporation. All rights reserved.
+//
+// Author: Marcin Maka <marcin.maka@linux.intel.com>
+
+#include <xtensa/coreasm.h>
+#include <xtensa/cacheattrasm.h>
+#include <cavs/lp_ctx.h>
+
+#define MEMCTL_INIT_BIT 23
+#define MEMCTL_INIT_VAL (MEMCTL_L0IBUF_EN | 1 << MEMCTL_INIT_BIT)
+#define SW_INT_NUM 7
+#define SW_INT_MASK (1<<SW_INT_NUM)
+
+#define L1_CACHE_PREFCTL 0x0000
+
+	.section .lps_vector, "ax"
+lps_pic_restore_vector_literals:
+	.literal_position
+	.align 4
+
+	.macro restore_vector	level
+		l32i	a4, a3, lp_ctx_vector_level_&level
+		wsr	a4, EXCSAVE+\level
+	.endm
+
+	.global lps_pic_restore_vector_literals
+	.global lps_pic_restore_vector
+	.global lps_pic_restore_vector_end
+
+lps_pic_restore_vector:
+	movi a0, 0
+	movi a1, 1
+	wsr  a1, WINDOWSTART
+	wsr  a0, WINDOWBASE
+	rsync
+
+	movi a3, MEMCTL_INIT_VAL
+	wsr a3, MEMCTL
+	rsync
+
+	movi    a3, 0x15
```

Comment:
Maybe put this magic 0x15 behind some meaningful define?

Reply:
Explanation added.